### PR TITLE
thread test date all the way through

### LINKED ISSUE #882: Store and return testDate on verify code API
### TL;DR

Mobile apps would like to differentiate between testDate and symptomDate for purposes of prompting the user.

Currently this is collapsed to a single date that is stored and transported through the verify response, verification token, and verification certificate.

### Design

**Proposal**

1) add `TestDate` to the verification code struct
2) store this in the DB
3) when the verificaiton token is minted, expand the subject to contain `diagnosis.symptomdate.testdate`
4) verification token structure will remain the same - a single date called symptom date, that can be either.

/kind enhancement

### PR DESCRIPTION
**Pull request by mikehelmick**
Fixes #882

## Proposed Changes

* Add test date into verify API return
* Thread this all the way through the verification token

**Release Note**

```release-note
API CHANGE : /api/verify now returns the testDate in addition to symptom date if present. When the verification certificate is issue only one interval is inserted: symptomDate if present, testDate if not.
```

### CODE REVIEW

**Comment by mikehelmick** [comment]
/hold
for verification w/ clinical team

File: pkg/database/token.go [hunk 1]
```diff
@@ -77,8 +85,8 @@ func (s *Subject) SymptomInterval() uint32 {
 
 func ParseSubject(sub string) (*Subject, error) {
 	parts := strings.Split(sub, ".")
-	if length := len(parts); length < 1 || length > 2 {
-		return nil, fmt.Errorf("subject must contain 2 parts, got: %v", length)
+	if length := len(parts); length < 2 || length > 3 {
```
Comment (sethvargo) [inline pkg/database/token.go:88]: Won't think break existing JWT tokens that have the old subject?

Comment (mikehelmick) [reply pkg/database/token.go:88]: the old if statement was wrong - it should have always been 2 parts before.

Why I didn't write `length != 2` the first time, we'll never know.

Comment (jeremyfaller) [reply pkg/database/token.go:88]: Not that I doubt your code, but this feels like a test.

Comment (sethvargo) [reply pkg/database/token.go:88]: ^ what Jeremy said

Comment (mikehelmick) [reply pkg/database/token.go:88]: funny enough - the 2 segment subject was the only thing under test before.
test coverage has been boosted to cover all scenarios.

**Comment by jeremyfaller** [comment]
/lgtm

still hold, pending clinicians.

**Pull request merged**